language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1998,4 +2000,94 @@ public static Optional<Node> getMatchingNode(Node currentNode, Predicate<Node> p
+     */
+    public static List<String> getFuncArguments(FunctionSymbol symbol, BallerinaCompletionContext ctx) {
+        List<String> args = new ArrayList<>();
+        Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> params = CommonUtil.getFunctionParameters(symbol, ctx);
+        List<ParameterSymbol> reqParams = params.getLeft();
+        for (ParameterSymbol param : reqParams) {
+            args.add(getParamTypeAndNamePair(param, ctx).orElse(""));
+        }
+        if (params.getRight().isPresent()) {
+            args.add(getParamTypeAndNamePair(params.getRight().get(), ctx).orElse(""));
+        }
+        return args;
+    }
+
+    /**
+     * Get the parameter type and name pair given the paramter symbol.
+     *
+     * @param param parameter symbol.
+     * @param ctx   Lang Server Operation context
+     * @return {@link Optional<String>} Type and name pair.
+     */
+    public static Optional<String> getParamTypeAndNamePair(ParameterSymbol param, BallerinaCompletionContext ctx) {
+
+        if (param.paramKind() == ParameterKind.REST) {
+            ArrayTypeSymbol typeSymbol = (ArrayTypeSymbol) param.typeDescriptor();
+            return Optional.of(CommonUtil.getModifiedTypeName(ctx, typeSymbol.memberTypeDescriptor())
+                    + (param.getName().isEmpty() ? "" : "... "
+                    + param.getName().get()));
+        }
+
+        if (param.typeDescriptor().typeKind() == TypeDescKind.COMPILATION_ERROR) {
+            // Invalid parameters are ignored, but empty string is used to indicate there's a parameter
+            return Optional.empty();
+        } else {
+            return Optional.of(CommonUtil.getModifiedTypeName(ctx, param.typeDescriptor()) +
+                    (param.getName().isEmpty() ? "" : " " + param.getName().get()));
+        }
+    }
+
+    /**
+     * Get the list of function parameters from the invokable symbol.
+     *
+     * @param symbol Invokable symbol to extract the parameters
+     * @param ctx    Lang Server Operation context
+     * @return Pair of list of parameter symbols and rest parameter symbol.
+     */
+    public static Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> getFunctionParameters(
+                    FunctionSymbol symbol, BallerinaCompletionContext ctx) {
+        boolean skipFirstParam = skipFirstParam(ctx, symbol);
+        FunctionTypeSymbol functionTypeDesc = symbol.typeDescriptor();
+        Optional<ParameterSymbol> restParam = functionTypeDesc.restParam();
+        List<ParameterSymbol> parameterDefs = new ArrayList<>();
+
+        if (functionTypeDesc.params().isPresent()) {
+            List<ParameterSymbol> params = functionTypeDesc.params().get();
+            if (skipFirstParam) {
+                if (params.size() > 1) {
+                    parameterDefs.addAll(params.subList(1, params.size()));
+                }
+            } else {
+                parameterDefs.addAll(params);
+            }
```

Comment:
`parameterDefs.addAll(params.subList(skipFirstParam ? 1 : 0, params.size()))`

What about something like the above? Would that satisfy the requirement?

Reply:
The suggested code snippet won't work as we have to check for the size of the params before slicing. Adding that condition into a conditional expression will further complicate the logic IMO.